crates/rethnet_evm/benches/state.rs

```diff
@@ -0,0 +1,131 @@
+use criterion::{criterion_group, criterion_main, BatchSize, Criterion};
+use revm::{db::StateRef, primitives::AccountInfo};
+use std::{clone::Clone, str::FromStr};
+
+use rethnet_eth::Address;
+use rethnet_evm::state::{HybridState, LayeredState, RethnetLayer, StateError, SyncState};
+
+#[derive(Default)]
+struct RethnetStates {
+    layered: LayeredState<RethnetLayer>,
+    hybrid: HybridState<RethnetLayer>,
+}
+
+impl RethnetStates {
+    fn fill(&mut self, number_of_accounts: u64, number_of_layers_per_account: u64) {
```

Comment:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3868/commits/b9040fb8bc7ac3b1b369ac4889d003e564c3db76

Reply:
This might be the perfectionist in me, but this approach has two caveats:
1. Not all checkpoints have an equal number of accounts
2. It's more costly to set up

Alternatively, you could do

```rust
for _ in 0..number_of_checkpoints {
  for i in 1..=number_of_accounts_per_checkpoint {
    ..
  }
}
```

which avoids the `%` operator and has an equal number of accounts per checkpoint

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3868/commits/b9040fb8bc7ac3b1b369ac4889d003e564c3db76

Reply:
Actually that latest change is no good. it drastically increased the test setup time (like over 90 minutes just for the `insert` benchmark). I need to look further at this, so moving the PR back to Draft status.

Reply:
It was a simple mistake, fixed in https://github.com/NomicFoundation/hardhat/pull/3868/commits/9b6d427b5526625bd786b6e0a84fa58af14d6ea4